### PR TITLE
Read menu page categories from Lettera

### PR DESCRIPTION
**Pull request by vapaj**
It seems or some categories, for instance teater, Lettera will not return anything. The queries must be fixed there.

### CODE REVIEW

File: apps/mosaico/src/Mosaico/Header/Menu.purs [hunk 1]
```diff
@@ -116,103 +109,17 @@ render { props: { visible } } = DOM.div
...
-                       }
-                     ]
-                 }
+    mkSection (n /\ acc) (Category c) =
+      let mkSubsection (Category subc) =
+            { title: subc.label, url: "/" <> show subc.label }
+          section =
+            Section $
+              { title: toUpper $ unwrap c.label
+              , modifier: Just $ "--section" <> show n
```
Comment (vapaj) [inline apps/mosaico/src/Mosaico/Header/Menu.purs:118]: Not sure if the `--section<n>` is used anywhere at this point for the categories, but maybe better to leave in if required by the `Section` type.

Comment (ae-mo) [reply apps/mosaico/src/Mosaico/Header/Menu.purs:118]: I'm pretty sure it's a remnant of a previous version of the menu. I don't see any modifier mentioned in the `menu.less` file for sections.

Also I think that reading the category structure from the server makes it impossible to have ad hoc modifiers without coupling the server and client implementations, in the sens that the server would become aware of the layout that the client wants to render. All this to say that maybe it makes sense to remove that `modifier` field from the data structure :rofl: 
Unless we come up with some non ad hoc modifier we might want to attach on the fly but I can't come up with such an example now.

Comment (vapaj) [reply apps/mosaico/src/Mosaico/Header/Menu.purs:118]: Let's remove it 😁

**Comment by ae-mo** [comment]
Related to #771

File: apps/mosaico/src/Mosaico/Header/Menu.purs [hunk 2]
```diff
@@ -253,8 +152,8 @@ render { props: { visible } } = DOM.div
         renderMenuLayoutElement (Separator modifier) = renderSeparator modifier
 
         renderSection :: Section -> JSX
-        renderSection { modifier, subsections, title } = DOM.div
-          { className: unwords [ sectionClass, sectionClass <> fromMaybe mempty modifier ]
+        renderSection { subsections, title } = DOM.div
+          { className: unwords [ sectionClass, sectionClass ]
```
Comment (ae-mo) [inline apps/mosaico/src/Mosaico/Header/Menu.purs:156]: The second `sectionClass` element is redundant

File: apps/mosaico/src/Mosaico/Header.purs [hunk 3]
```diff
@@ -108,7 +108,13 @@ render { state: { menuVisible, menuComponent }, props } =
                  else
                    [ searchButton ])
                 <> [ DOM.div_
-                      [ menuComponent { visible: menuVisible }
+                      [ menuComponent
```
Comment (ae-mo) [inline apps/mosaico/src/Mosaico/Header.purs:111]: I think we can get rid of this instance of the menu component here, as this is a leftover from back when the menu was still an overlay.

**Review by ae-mo — CHANGES_REQUESTED** [review]

File: apps/mosaico/src/Mosaico/Header.purs (hunk 3 above)
Comment (vapaj) [reply apps/mosaico/src/Mosaico/Header.purs:111]: I see 🤔 I'll do a cleanup

Comment (vapaj) [reply apps/mosaico/src/Mosaico/Header.purs:111]: I guess all the `menuVisible` stuff as well

**Review by ae-mo — APPROVED** [review]

**Pull request merged**